java/ql/lib/semmle/code/java/Statement.qll

```diff
@@ -428,6 +450,20 @@ class SwitchCase extends Stmt, @case {
     result = this.getSwitch().getExpr() or result = this.getSwitchExpr().getExpr()
   }
 
+  /**
+   * Gets the `i`th case in this case's switch block.
+   */
+  SwitchCase getSiblingCase(int i) {
```

Comment:
Are we sure that we really want this as a general-purpose convenience predicate? It's inherently quadratic, so we may very well end up in a situation where we're forced to deprecate it for performance reasons.

Reply:
Suggest I follow up immediately after this PR to introduce `class SwitchBlock` and thereby remove everything shared between SwitchStmt and SwitchExpr.